data/transactions/logic/eval.go

```diff
@@ -975,6 +1006,7 @@ func EvalContract(program []byte, gi int, aid basics.AppIndex, params *EvalParam
 			}
 		}
 		cx.readBudgetChecked = true
+		cx.SurplusReadBudget = cx.ioBudget - used
```

Comment:
That is actually already supported (even before my most recent PRs, which changed it a bit).

The code here checks to see if it's possible to read in this box given the max possible read budget, calculated from the remaining unnamed references we have available, plus `cx.SurplusReadBudget`, the unused read budget from "normal" execution: https://github.com/algorand/go-algorand/blob/7de6599a945ca9d89071bd11bf50f64c2ed95b89/ledger/simulation/resources.go#L379-L388

Reply:
I was using `AddSaturate` and friends a little too much, I've simplified in 50fa7ef0d

I'm still going to use it for computing `usedReadBudget`, since the `logic` package code also uses it when calculating read budget usage, but it's likely not necessary.

Reply:
ok! resolving